commandsv3/src/main/java/org/wpilib/commands3/Mechanism.java

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package org.wpilib.commands3;
+
+import edu.wpi.first.units.measure.Time;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Generic base class to represent mechanisms on a robot. Commands can require sole ownership of a
+ * mechanism; when a command that requires a mechanism is running, no other commands may use it at
+ * the same time.
+ *
+ * <p>Even though this class is named "Mechanism", it may be used to represent other physical
+ * hardware on a robot that should be controlled with commands - for example, an LED strip or a
+ * vision processor that can switch between different pipelines could be represented as mechanisms.
+ */
+public class Mechanism {
+  private final String m_name;
+
+  private final Scheduler m_registeredScheduler;
+
+  /**
+   * Creates a new mechanism registered with the default scheduler instance and named using the name
+   * of the class. Intended to be used by subclasses to get sane defaults without needing to
+   * manually declare a constructor.
+   */
+  @SuppressWarnings("this-escape")
+  protected Mechanism() {
+    m_name = getClass().getSimpleName();
+    m_registeredScheduler = Scheduler.getDefault();
+    setDefaultCommand(idle());
```

Comment:
Is there any reason for this to not be a delegate constructor?
```suggestion
    this(getClass().getSimpleName());
```

Reply:
Huh, TIL. That's an annoying Java limitation.